Modules/System/Cryptography Management/src/X509Certificate2Impl.Codeunit.al

```diff
@@ -0,0 +1,120 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+
+codeunit 1285 "X509Certificate2 Impl."
+{
+    Access = Internal;
+
+    var
+        CertInitializeErr: Label 'Unable to initialize certificate!';
+
+    procedure VerifyCertificate(var CertBase64Value: Text; Password: Text; ContentType: Enum "X509 Content Type"): Boolean
+    var
+        X509Certificate2: DotNet X509Certificate2;
+    begin
+        if not TryInitializeCertificate(CertBase64Value, Password, X509Certificate2) then
+            Error(CertInitializeErr);
+        if not TryExportToBase64String(X509Certificate2, ContentType, CertBase64Value) then
+            Error(GetLastErrorText());
+        exit(true);
+    end;
+
+    procedure GetCertificateFriendlyName(CertBase64Value: Text; var FriendlyName: Text)
```

Comment:
All the functions that uses CreateX509Certificate will only work if the cert doesn't have a password. 
For example if you try to call 
GetCertificateFriendlyName(certBase64, FriendlyNameTxt);
You will get an error that you can't read the cert without a password. at least with a pfx cert

Reply:
Idea was to verify cert before getting any property value from cert, but I realize that wasn't best solution